node/tessel.js

```diff
@@ -628,4 +636,72 @@ var SPISettings = {
     CPHA: 2
 };
 
+function LED(color, path) {
+    this.color = color;
+    this._path = path;
+}
+
+LED.prototype.high = function(callback) {
+    this.write(true, callback);
+}
+
+LED.prototype.low = function(callback) {
+    this.write(false, callback);
+}
+
+LED.prototype.output = function(value, callback) {
+    this.write(value, callback);
+}
+
+LED.prototype.input = function(callback) {
+    if (typeof callback === 'function') {
+        callback(new Error("Cannot make LEDs an input."));
+    }
+}
+
+LED.prototype.toggle = function(callback) {
+    var self = this;
+    self.read(function(err, value) {
+        if (err) {
+            if (typeof callback === 'function') {
+                callback(err);
+            }
+            return;
+        }
+        else {
+            self.write(!value, callback);
+        }
+    })
+}
+
+LED.prototype.write = function(value, callback) {
+    fs.writeFile(this._path, value ? '1' : '0', callback);
+}
+
+LED.prototype.read = function(callback) {
+    fs.readFile(this._path, function(err, value) {
+        if (typeof callback !== 'function') {
+            return;
+        }
+        if (err) {
+            callback(err);
+            return;
+        }
+        else {
+            value = value.toString().trim();
+            if (value === '1') {
+                callback(null, true);
+                return
+            }
+            else if (value === '0') {
+                callback(null, false);
+                return
+            }
+            else {
+                throw new Error("Invalid state returned by LED:" + value);
```

Comment:
> I'm sorry to be such a pain in the arse but I think it's important that best practices be followed in such a large project as this :)

No apologies necessary! Please continue to point out questionable code - it helps the project and the programmers (like myself!) improve.

> In what way is it not a recoverable error? 

I was on the fence about this because it seems like if you can't use an LED for visual indication as expected, you are out of luck. I suppose you can turn on another LED to indicate that something went wrong. 

In any case, I think you're right that, for consistency sake, this should be provided in the callback. Thanks for looking through this! Another PR coming soon.